lua/neotest-bash/core/spec_builder.lua

```
@@ -1,14 +1,21 @@
 local root_finder = require("neotest-bash.core.root_finder")
+local bashunit_finder = require("neotest-bash.util.bashunit_finder")
 local CommandBuilder = require("neotest-bash.util.command_builder")
 
 SpecBuilder = {
 	---@param args neotest.RunArgs
+	---@param config? neotest-bash.AdapterConfig|nil
 	---@return nil | neotest.RunSpec | neotest.RunSpec[]
-	build_spec = function(args)
+	build_spec = function(args, config)
 		local tree = args.tree
 		local tree_data = tree:data()
 		local path = tree_data.path
 		local root = root_finder.findRoot(tree_data.path)
+		local bashunit_path = bashunit_finder.findBashunit(root, config)
+
+		if not bashunit_path then
+			return {}
```

I would throw an error with the message for the user here, so the user understand that they need the binary.